src/actions/generalActions.js

```diff
@@ -0,0 +1,57 @@
+const reactToGeneralAction = (model, toolHandler) => {
+	return (action) => {
+		switch (action.type) {
+			case 'MODE_UPDATE':
+				if (model.activityInProgress === true) {
+          toolHandler.cleanTool();
```

Comment:
add comment here on when does this get invoked, what are the possible scenarios in which activityInProgress is true, and what will cleanTool do.

Reply:
I'm planning on turning the model defined in createModel.js into a class, with class functions that can alter model's state. Agreed that changing model params inside the actions.js get very confusing.